### PR TITLE
add digitalocean_droplets data source

### PR DESCRIPTION
**Pull request by tdyas**
Adds a `digitalocean_droplets` data source.

Note: I don't have access to the VPC preview so the existing tests for the `digitalocean_droplet` data source fail when I ran those acceptance tests to ensure that the refactoring didn't cause a regression.

### CODE REVIEW

**Comment by tdyas** [comment]
> Note: I don't have access to the VPC preview so the existing tests for the digitalocean_droplet data source fail when I ran those acceptance tests to ensure that the refactoring didn't cause a regression.

To clarify, some of the acceptance tests use `digitalocean_vpc` in setting up droplets.

File: digitalocean/droplets.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		"vcpus":         droplet.Vcpus,
+		"memory":        droplet.Memory,
+		"status":        droplet.Status,
+		"locked":        droplet.Locked,
+		"created_at":    droplet.Created,
+		"vpc_uuid":      droplet.VPCUUID,
+	}
+
+	if droplet.Image.Slug == "" {
+		flattenedDroplet["image"] = droplet.Image.ID
```
Comment (andrewsomething) [inline digitalocean/droplets.go:171]:
```suggestion
		flattenedDroplet["image"] = strconv.Itoa(droplet.Image.ID)
```


Working with both the droplet and droplets data sources, I'm running into:

```
Error: Unable to set `image` attribute: image: '' expected type 'string', got unconvertible type 'int'

  on main.tf line 8, in data "digitalocean_droplet" "drop":
   8: data "digitalocean_droplet" "drop" {
```

File: website/docs/d/droplets.html.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  filter {
+    key = "size"
+    values = ["s-1vcpu-1gb"]
+  }
+  filter {
+    key = "backups"
+    values = ["true"]
+  }
+  sort {
+    key = "created"
```
Comment (andrewsomething) [inline website/docs/d/droplets.html.md:48]:
```suggestion
    key = "created_at"
```

File: website/docs/d/droplets.html.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+# digitalocean_droplets
+
+Get information on Droplets for use in other resources, with the ability to filter and sort the results.
+If no filters are specified, all Droplets will be returned.
+
+This data source is useful if the Droplets in question are not managed by Terraform or you need to
+utilize any of the Droplets' data.
+
+Note: You can use the [`digitalocean_droplet`](/docs/providers/do/d/droplet.html) data source to obtain metadata
+about a single image if you already know the `id`, unique `name`, or unique `tag` to retrieve.
```
Comment (andrewsomething) [inline website/docs/d/droplets.html.md:18]:
```suggestion
about a single Droplet if you already know the `id`, unique `name`, or unique `tag` to retrieve.
```

Comment (tdyas) [reply website/docs/d/droplets.html.md:18]: Thanks, good catch! Obviously, you can tell what I and Copy McPastington used as a template. :)

**Comment by tdyas** [comment]
Thanks for testing this!

**Pull request merged**